### ISSUE TITLE
ILoaderOptions xhrType should be 'number' instead of 'string'

### ISSUE THREAD

**Issue by qtiki** [pixijs/pixi-typescript#179]
https://github.com/pixijs/pixi-typescript/blob/19d8cdeb80539f4584d31e2acb2c6e5316473efd/pixi.js.d.ts#L2385

Title says it all.

**Comment by clark-stevenson** [comment]
@tribetiki 

I am not sure!

[It](https://github.com/englercj/resource-loader/blob/master/src/Loader.js#L251) refers to a `Resource.XHR_RESPONSE_TYPE` which [seems to be a string?](https://github.com/englercj/resource-loader/blob/master/src/Resource.js#L985)

**Comment by ivanpopelyshev** [comment]
It should be enum.

**Comment by qtiki** [comment]
@clark-stevenson Looks like you're right. I am a bit baffled how I ended up getting a TypeScript error from that in the first place (which is why I opened this issue). Now that I removed my "as any" workaround it seems to work without problems. The npm modules have been updated after I first had the problem so it might be because of that, or then I'm just slowly losing my mind.

**Comment by qtiki** [comment]
Just to prove I'm not completely bonkers (mainly to myself) I dug around a bit more:

We are using pixi-spine which has a dependency to types/pixi.js exact version 4.3.1. Basically this means our root project is using the latest types but we have another set of (older) types under pixi-spine. In that version of the types the XHR_RESPONSE_TYPE is defined like this:

``` typescript
static XHR_RESPONSE_TYPE: {
    DEFAULT: number;
    BUFFER: number;
    BLOB: number;
    DOCUMENT: number;
    JSON: number;
    TEXT: number;
};
```
This is of course incompatible with the xhrType?: string for the LoaderOptions. It seems that Visual Studio Code is semi-randomly using the types from pixi-spine instead of the ones from the root so the type error creeps in every now and then.

TL;DR - The problem is already fixed in pixi types but pixi-spine is using old types! (Maybe I'll go open an issue there...) - Case closed!

**Comment by ivanpopelyshev** [comment]
@tribetiki we've just updated pixi-spine, going to upload new version on npm. 

UPD: 1.5.5 uploaded.

**Comment by qtiki** [comment]
@ivanpopelyshev I just updated pixi-spine to 1.5.5 and I can confirm that the issue I described is fixed!

Thank you both very much. The world is now a better place. 🦄 🌈 ⭐️
